### PR TITLE
Olwa pr

### PR DESCRIPTION
**Pull request by oliver-walker-hms**
last OSIsoft change for a while

### CODE REVIEW

File: src/com/hms/flexyosisoftconnector/OSIsoftConfig.java [hunk 1]
```diff
@@ -357,6 +361,23 @@ public static void setFlexyName() {
     flexyName = res;
   }
 
+  /** Read the Flexy serial number and return it. */
```
Comment (alexjhawk) [inline src/com/hms/flexyosisoftconnector/OSIsoftConfig.java:364]: Javadoc doesnt make sense

File: src/com/hms/flexyosisoftconnector/OSIsoftConfig.java [hunk 2]
```diff
@@ -357,6 +361,23 @@ public static void setFlexyName() {
...
+      SysControlBlock sysControlBlock = new SysControlBlock(SysControlBlock.INF);
+      flexySerialNum = sysControlBlock.getItem("SerNum");
+    } catch (EWException e) {
+      Logger.LOG_SERIOUS(
+          "An error occurred while getting the serial number of the host Ewon Flexy.");
+      Logger.LOG_EXCEPTION(e);
+    }
+  }
+
+  /** */
```
Comment (alexjhawk) [inline src/com/hms/flexyosisoftconnector/OSIsoftConfig.java:376]: missing javadoc

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: src/com/hms/flexyosisoftconnector/OSIsoftConfig.java (hunk 1 above)
Comment (oliver-walker-hms) [reply src/com/hms/flexyosisoftconnector/OSIsoftConfig.java]: Java docs are usually the last thing on my mind. Fixed

File: src/com/hms/flexyosisoftconnector/OSIsoftConfig.java [hunk 3]
```diff
@@ -357,6 +361,27 @@ public static void setFlexyName() {
...
+      Logger.LOG_SERIOUS(
+          "An error occurred while getting the serial number of the host Ewon Flexy.");
+      Logger.LOG_EXCEPTION(e);
+    }
+  }
+
+  /**
+   * Get the Flexy serial number.
+   *
+   * @return returns the Flexy serial number
```
Comment (TomKimsey) [inline src/com/hms/flexyosisoftconnector/OSIsoftConfig.java:379]: You dont need to specify returns in an @return.

File: src/com/hms/flexyosisoftconnector/PayloadBuilder.java [hunk 4]
```diff
@@ -428,8 +428,16 @@ public static String getContainerSettingJson(int startTagIndex, int numToProcces
           payload.append(",");
         }
 
+        String flexySerial = OSIsoftConfig.getFlexySerial();
```
Comment (TomKimsey) [inline src/com/hms/flexyosisoftconnector/PayloadBuilder.java:431]: final?

File: src/com/hms/flexyosisoftconnector/PayloadBuilder.java [hunk 5]
```diff
@@ -428,8 +428,16 @@ public static String getContainerSettingJson(int startTagIndex, int numToProcces
...
+
+        /* Ensure the container is uniquely identified across devices communicating to the same
+         * OSIsoft server under the same Type declaration.
+         * Types are treated as immutable and changing the type creates a new container.
+         */
+        String containerId = tagName + "-" + typeString + "-" + flexySerial;
```
Comment (TomKimsey) [inline src/com/hms/flexyosisoftconnector/PayloadBuilder.java:437]: final?

File: docs/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,11 @@
 # Flexy OSIsoft Changelog
 
+## v2.5
+### Major Changes
++ None
+### Minor Changes
+OMF container ID's are unique across devices
```
Comment (TomKimsey) [inline docs/CHANGELOG.md:7]: missing +

**Review by TomKimsey — CHANGES_REQUESTED** [review]

File: src/com/hms/flexyosisoftconnector/OSIsoftConfig.java [hunk 7]
```diff
@@ -357,6 +361,27 @@ public static void setFlexyName() {
     flexyName = res;
   }
 
+  /** Read the Flexy serial number and store it. */
+  public static void setFlexySerial() {
```
Comment (alexjhawk) [inline src/com/hms/flexyosisoftconnector/OSIsoftConfig.java:365]: I think this should probably be #readFlexySerial(). setFlexySerial() makes it seem like the serial number is mutable.

**Review by alexjhawk — CHANGES_REQUESTED** [review]

**Review by TomKimsey — APPROVED** [review]

**Review by alexjhawk — APPROVED** [review]

**Pull request merged**